src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddleware.cs

```diff
@@ -143,6 +143,8 @@ private async Task HandleException(HttpContext context, ExceptionDispatchInfo ed
                 }
 
                 _logger.ErrorHandlerNotFound();
+
+                edi = ExceptionDispatchInfo.Capture(new InvalidOperationException($"No exception handler was found, see inner exception for details of original exception. If an exception should not be thrown for 404 responses, set {nameof(ExceptionHandlerOptions.AllowStatusCode404Response)} to true.", edi.SourceException));
```

Comment:
Who do we think this new exception is for? We don't expect there to be any application code above this in the pipeline, it will be the servers that catch this and log it again as an error. Does that make for redundant logs? What does the log output look like now for this scenario on Kestrel?

Reply:
I think the spirit of the recommendation [here](https://github.com/dotnet/aspnetcore/issues/31024#issuecomment-803111390) is that logs are easily missed whereas exception message is more explicit and potentially useful to the user. However, I do agree that in most cases, there are many levels of duplicate logging. Even the log message and the exception message are similar. 

I was debating whether to wrap this exception or not but I figured I'll illustrate what wrapping looks like and we can decide whether the change is beneficial or not.

Reply:
In theory exceptions are more noticeable, but in practice I don't know that there's any user code up the stack to notice this one, only the server.

Reply:
The server logs this as an error which is more noticeable. I might even recommend just throwing and not logging if the warning didn't already exist.

Reply:
That makes sense as well, but in that case I would put the message that's currently logged into the exception. It feels quite long for an exception message though.